packages/echo-base/src/errors/BaseError.ts

```diff
@@ -123,10 +123,13 @@ export function findPropertyByName(
  * Supports custom Error which has public or private fields. Will ignore any functions on the Error object.
  * It does not support cycling references (A -> B -> A).
  * @param objectWithProperties an object containing properties
+ * @param args.ignoreEquals ignore/omit any property that equals this (case insensitive)
+ * @param args.ignoreIncludes ignore/omit any property that includes this (case insensitive)
  * @returns a record containing all properties of the object
  */
 export function getAllProperties(
-    objectWithProperties: Record<string, unknown> | Error | undefined
+    objectWithProperties: Record<string, unknown> | object | undefined,
+    args?: { ignoreEquals?: string[]; ignoreIncludes?: string[] }
```

Comment:
Based on the code it seems that `ignoreEquals` and `ignoreIncludes` are mutually excluding each other.
But this type doesn't reflect that.

I think something like this would make a XOR operator on them:
```
type MyType = { ignoreEquals: string[]; ignoreIncludes?: never } | { ignoreEquals?: never; ignoreIncludes: string[] };

const myVar: MyType = { ignoreEquals: [ 'a', 'b', 'c' ] };
const myVar2: MyType = { ignoreIncludes: [ 'e', 'f', 'g' ] };
const myVar3: MyType = { ignoreIncludes: [ 'a', 'b', 'c' ], ignoreEquals: [ 'x' ] }; // type error: can't use both at the same time
```

Reply:
They are not mutually excluding each other, added a unit test as an example.